### ISSUE TITLE
Fix `function-no-unknown` false positives for unspaced operators against nested brackets

### ISSUE THREAD

**Issue by IgnatovDan** [stylelint/stylelint#6582]
### What steps are needed to reproduce the bug?

Add a css file with this class:
```
.class1 {
  height: calc(10 - (10 / 10*(10 - 1)));
  height: calc(10 - (10 / (10 + 1)*(10 - 1)));
}
```

### What Stylelint configuration is needed to reproduce the bug?

```
module.exports = {
  "rules": {
    "function-no-unknown": true,
  }
}
```

### How did you run Stylelint?

"stylelint": "npx stylelint \"**/*.css\"",

### Which version of Stylelint are you using?

14.16.0

### What did you expect to happen?

No problems to be reported.

### What actually happened?

```
  8:27  ✖  Unexpected unknown function "10*"    function-no-unknown
  9:33  ✖  Unexpected unknown function "*"      function-no-unknown
```

### Does the bug relate to non-standard syntax?

No

**Comment by jeddy3** [comment]
@IgnatovDan Thanks for the report and for using the template.

I [can reproduce](https://stylelint.io/demo#N4Igxg9gJgpiBcICGACYAdAdilALGAlgOa4Au8KYSANmABQCMADAA4AeAVHQKxfMoBaFNwCUYgNxYAvlhAAaEAGcAnplJI2CcIsXzwETADNiWjNhToQAJwCu1GIssUzOHJaSZMEdaQIGBXgI2mADWXgDumE4opLYwcliuFsikArb2ARBBoRFRCDFxCeZuIABG1BBgIZkCMAC2LKTK0bE28YmulpAVVjUEmABuNARQAvhsLYUdJZB1dTBqNfWNzfmt7cXJYDaKpBB1AixWECwwVk01dQQ6-UQCQ72GwWC+BpNtRUmWsGDUSFZIV6YATlSrVQJQGwsagEKikGACba7faHY6nc4EBzvDZfEA-P4AoEgipVGqQ6GwwEIo4nM6+LH5ADa01c60+STQLNxxC8Vhg0WZmw5WwMihg218AwR5JhcIcAnCBFIuAEUAIhkMZwWqSG1DajhAXNcAF0jTJNsb2SV8f9AX5gaDSYFFLgIOdcB5RjT0RcIFKrFYRgyKGyWZZDAZUoYkFdqMoyVDZVSAjHgwUPmGQBHFtHY-HAlcbpg7kQFmdYQJcwQ4wIQjBlOE3VBsVbkk9MC97YiaGAasFFCwkGAYKNaYS3S3M+3O-4YZgYP8BEQAWrtTUvJhdp7-qM1XyZ3kQ1NNuHnkTAsEwhBIpOTyA68pDAD5sSwQmKXKBGL7C83Qajxmd4Pk+qaquKBJ2v4gQEA0brqGot64vMapIJWC6kDYfIpi+F45Neh7pjinQgJgqajMuIwCP8C6KH0gzDM2azHriEIOMOmBqsWX6nGA6qwkqqwAURJQQomlLwlRqQwSwcFpHYaahneokfsm37insVj-oRraWIEywXIoECYcOiHEXpbCxKhYpXN0oqmSJWT9LqlFQEZ5QIoofwuoi+zzGoWmKSxjn0TCXoQIoSpdoCAjSbJ6T8kxgFBTFAYwEQdiLuEuBKg4g4mYlwnJN6dL5lkl65PZyRqb+vQsGKNiuYiXm0bhV43gVOlKDAP4aYc9WNd19Rrq1FUdZm1W9U0px9nh7UUIKwqBRyLjCskPJugl81GriSJ7Acg1+aQBrbSgppCig5pJJa42xLc64wOEc6bdpmbBEqM1tQRS0lDQLAevcNBtJkPj2gKRqWOi7HqKWlitjgK3CpYMBsMOjQAAponSmJaQtq0lCcQ4CbDJ0lMY1DUAIBO8U0xPnbihgVNAlN5UT8gk1VewsMzhM02zdPEbsxx1tz1OrCdZ3CpdJqddF8W1A0FxPSCMARnyYPne41DhEgygGnDnL80jKMwI06t4yUjohPYOhUYY8K9IooGkS+lvW3r7PhgQmmpPOuwjpY4v6yU61q0yHvIHbZw+XM2oB-zEsclLODXXesvydhCJUGK0SWBU4RnLTuJpxkzseXlCJIJHVg58gWs67REXFhkT2F8RxcIoEUocW6qIqwQExjanqRy9ZsIQBUwLzo9-QV1XNc0NruutxbJLVL84V3aUALDvLKwCErpQqxtNfzv6y-JJbTUb1xW9Dh3D1K5X9vz3XS987il-r43dy3zvU8H0fUOFBNaL1onUOwvh94z3PpYT+FRv4gm3mXO+ytVbPRAfXL8txm7QPfsRS+tJMCbyQQEB+M9bbP3yBg3WAhwHUEgS3PBK83yEOISggcKCn4FyobXUBWCm4IkYZ1AhpwiE3xIRwneh80Ev1ATA-QPRKxni7F4EGbweF1GgGcPInVbK9HGN2bOPC87cN0ePHuBj7DFmVDXF0cF5GzEOrvRW5DpHHzDhrXh9dz7wxOkbVGpAzbm09t7UhftGKBz8SAEOz1cZ40sLsZQ9g5ypEcZ6Y68czQshTjtXya4so5UkQifoEVYCyO8UwrYOw9qojHBcfS8YAEyI8YjLxb99YIw5P4k2gSWnBIjvbRE1SUTFXOKsIOxFjChN9vCCJ8cJmWBiUE+JAyo6ONjpU1pJSgz8KIDg+cr4qhxzxgnJISdTq6OGQcZCBBUKDlIPbAicTiIAD06CMlQgAL2NB8gQnymACAAJzGg4CIOgAhfn-KBcaAA1CIDgAASHxaBzk5OIrtEZmMxmHEBI85Zrz3lfJ+V8gFwLQXgshaS2F8KkWbJRdkzqNpCRdlKB4O4RSKHcOASRGAZ9KnfHAraIkrKuIcrcUA5IC8KmMsFcy-wl9R62UnmQg5XDq48KlTQuhDDcEyt+EKllq8vz1DHhPbinC54atfg3bBgjdWZiZZBB0RrFXmOBGKwB6CeV8r1RBYVLrbVQIOXUDQYF9Vys3NEBgvqDXyqNZZasd1XUT3Ke0h1sqnWHPBFkPkkIOIeFSBPEgnpak+mxpVAV4bM1KtIdPVVlruWarARAggQaEoxojT5M1HK1WpptQItt8jHVEhrR65p3LT6mPTVWokDS22oPcVtTxTbkWdNxMjAJ+LWlqrDX60GdL6Ze12GE2ZxzVpoq2UQXksTw47vWQhA9xFtmwF2fshElsz3ClOa4VFnVsxRhjNWfMoEACONhvBpmobRLKWo0jijyRxf2-KszKP8Gk2tj8G2SutbQltg7kPThHb5Ky5dOXqsbTh7+b75GEa7Ohsdi7kiTvVX+1DwIQ1sGcY0meWkmCsY7OeUCWcvUmJY1ONjOK+RqHwGKFqKrimbiDH23D9DW1CPEwJrsg4pPKgcPKDlz6vXMdfXa+cNGJOYQpmBiDWkoPmc0-4Ap8Ie1Ybs8h3AQNYDLhgB3bwTr54CPkbFc4wN-M8Ms0FxDagwsUAAEydWAs+Dyg0aqhaBDXSG2okClmyG7BUSpsrAgfI2Kwo5MA1g-chxLqYWqgXuXivpuI3mQuJX8ql5KIUkuhXCxFq7f2ZisUQZUfyzhZFwkqCtIAONcbbVpaNmZptK0GzY-IDBYt8YWyOW5aFASYUzm681O9e1Wrkchm5qFNS7awqO0j4qjO8qnXec7O2MJYVLoYkT15HtIS2xd9Ce2M41C7q5WqfJjADyEp1Z7l3XvUmotJvTDdSOGZPg9sTT3fsvYBwCYsCIxyAh7i5yhFHTtQ8xzD7HbK8fogJw7W7nq+3yOe2Bs43Hj3of-uQ47JPMHarU-ajHKEBAs6sGz1JxHDuz2J9hvhVHTPts20LkXYvo4hslwuiVuk0fyL0greMUzj2MMh5mPS48FTZWc+XSbBZrgIIWKOQwX4jJWD-vJ63Ng6iH16PYJAnE7ifNG4z5DmAPde9oaGo44prj7ooAAFk6iHz3UdAgJrnP70bAVmLEVGRcYTNdRPyJz6Ve49ue6R-B5NuWc6mmMeeT9HDfP8MdKiRunpW6ulZiPT7Bwp7DTzKiUsxrq13BV1Vxs8WWSLSdQmj3XFgZSg2Akr-OsR11co54cx+RM-ehz4IAviS+PepE65Ux7XyHt+SVuvv6nZxafq7u6jn140Uu9V39f4X4HnNB+ny-nu+qbb1baLt6WDNZEqUrQodbgHAo9a0odL9Z3gX6zClD9B37H7kYy7SrP49R-77DIGkRH707joYFpoIG-69CwCKDsRQAFrRx4F36BAbiS6TYX6USAFWBPLgwgCgF-KtZQpkpgqdZtbdY0p9YMpYHqQ9wwjs4S6c71rS5uY-7YHe7XDi5zAkYoIP4b5n6KESG9DTbV48bRAbakFKFA6l6g59wQ4vQmG6HAxpYx4igND2ADw6GpZ1QwANRZD-60R57GJfbo64gX7uGeFNRIAAHw66ayZr6KZlJaFP42FuH9RZAHRri+HcoF7n5kF9QeEDT2BOI1qqIxbJCuQ2DuRb5ZFTSZxhGfb5wBECyujuglpF41C5rBDUGLC6j6jMG3RcTWZf48IlFlHIZvSpLVH57+HyKdEIjFZNgfbjG1GTGAx2rSFqEYZc6uZtLNqqZN6ZhTFQIrFq5oHKZy47F3h7FSGqGHGEGMZa7xG4jnEqHh6cYGG+xGGdR7GdzmG9wV5D44DfSuBrpPpXobQABqSxOM4cpQEAA8E+50qKHQ5oUgIAUgQAA) using this valid property value:

```css
a { height: calc(10px*(5*(10 - 5))); }
```

It seems to be related to operators before _nested_ brackets. It may be an issue with [the upstream value parser we use](https://github.com/TrySound/postcss-value-parser). Please consider opening an issue there and, if applicable, resolving the problem upstream. In the meantime, you can workaround the issue by spacing your operators:

```css
a { height: calc(10px * (5 * (10 - 5))); }
```
